collector/perf_schema_replication_applier_status_by_worker.go

```diff
@@ -0,0 +1,207 @@
+
+// Scrape collects data from database connection and sends it over channel as prometheus metric.
+func (ScrapePerfReplicationApplierStatsByWorker) Scrape(ctx context.Context, db *sql.DB, ch chan<- prometheus.Metric) error {
+	perfReplicationApplierStatsByWorkerRows, err := db.QueryContext(ctx, perfReplicationApplierStatsByWorkerQuery)
+	if err != nil {
+		return err
+	}
+	defer perfReplicationApplierStatsByWorkerRows.Close()
+
+	var (
+		channelName, workerId                                                                   string
+		lastAppliedTransactionOriginalCommitTime, lastAppliedTransactionImmediateCommitTime     time.Time
+		lastAppliedTransactionStartApplyTime, lastAppliedTransactionEndApplyTime                time.Time
+		applyingTransactionOriginalCommitTime, applyingTransactionImmediateCommitTime           time.Time
+		applyingTransactionStartApplyTime                                                       time.Time
+		lastAppliedTransactionOriginalCommitSecond, lastAppliedTransactionImmediateCommitSecond float64
+		lastAppliedTransactionStartApplySecond, lastAppliedTransactionEndApplySecond            float64
+		applyingTransactionOriginalCommitSecond, applyingTransactionImmediateCommitSecond       float64
+		applyingTransactionStartApplySecond                                                     float64
+	)
+
+	for perfReplicationApplierStatsByWorkerRows.Next() {
+		if err := perfReplicationApplierStatsByWorkerRows.Scan(
+			&channelName, &workerId,
+			&lastAppliedTransactionOriginalCommitTime, &lastAppliedTransactionImmediateCommitTime,
+			&lastAppliedTransactionStartApplyTime, &lastAppliedTransactionEndApplyTime,
+			&applyingTransactionOriginalCommitTime, &applyingTransactionImmediateCommitTime,
+			&applyingTransactionStartApplyTime,
+		); err != nil {
+			return err
+		}
+
+		// Check if the value is 0, use a real 0
+		if lastAppliedTransactionOriginalCommitTime.Nanosecond() != 0 {
+			lastAppliedTransactionOriginalCommitSecond = float64(lastAppliedTransactionOriginalCommitTime.Unix())
+		} else {
+			lastAppliedTransactionOriginalCommitSecond = 0
+		}
+		ch <- prometheus.MustNewConstMetric(
+			performanceSchemaReplicationApplierStatsByWorkerLastAppliedTransactionOriginalCommitSecondDesc,
+			prometheus.GaugeValue, lastAppliedTransactionOriginalCommitSecond, channelName, workerId,
+		)
+
+		if lastAppliedTransactionImmediateCommitTime.Nanosecond() != 0 {
+			lastAppliedTransactionImmediateCommitSecond = float64(lastAppliedTransactionImmediateCommitTime.Unix())
+		} else {
+			lastAppliedTransactionImmediateCommitSecond = 0
+		}
+		ch <- prometheus.MustNewConstMetric(
+			performanceSchemaReplicationApplierStatsByWorkerLastAppliedTransactionImmediateCommitSecondDesc,
+			prometheus.GaugeValue, lastAppliedTransactionImmediateCommitSecond, channelName, workerId,
+		)
+
+		if lastAppliedTransactionStartApplyTime.Nanosecond() != 0 {
+			lastAppliedTransactionStartApplySecond = float64(lastAppliedTransactionStartApplyTime.Unix())
```

Comment:
Does MySQL return a more accurate time here? We typically use `float64(timeVal.UnixNano()) / 1e9` to get sub-second precision out of timestamps.